Content.Server/Storage/EntitySystems/CursedEntityStorageSystem.cs

```diff
@@ -0,0 +1,55 @@
+using Content.Server.Storage.Components;
+using Content.Shared.Audio;
+using Content.Shared.Interaction;
+using Robust.Server.Containers;
+using Robust.Shared.Audio;
+using Robust.Shared.Player;
+using Robust.Shared.Random;
+using System.Linq;
+
+namespace Content.Server.Storage.EntitySystems;
+
+public sealed class CursedEntityStorageSystem : EntitySystem
+{
+    [Dependency] private readonly IRobustRandom _random = default!;
+    [Dependency] private readonly EntityStorageSystem _entityStorage = default!;
+    [Dependency] private readonly ContainerSystem _container = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<CursedEntityStorageComponent, ActivateInWorldEvent>(OnActivate,
+            after: new[] { typeof(EntityStorageSystem) });
+    }
+
+    private void OnActivate(EntityUid uid, CursedEntityStorageComponent component, ActivateInWorldEvent args)
+    {
+        if (!TryComp<EntityStorageComponent>(uid, out var storage))
```

Comment:
Instead of listening for activation events, this should probably listen for EntityStorageClosed events. Then you wouldn't have to use ordered subscriptions.

Also you wouldn't create (presumably) unintentional interactions like: if a cursed locker has a id card lock, and someone tries to open it unsuccessfully, this will teleport the contents.